dev/src/index.ts

```diff
@@ -1007,6 +1025,21 @@ export class Firestore {
       });
   }
 
+  /**
+   * Terminates the Firestore client and closes all open streams.
+   * @return A Promise that resolves when the client is terminated.
+   */
+  async terminate(): Promise<void> {
+    await this._clientPool.terminate();
+    this._terminated = true;
```

Comment:
If you keep `_terminated`, you should flip these two statements. Otherwise, an operation can run during termination.

Reply:
oops, thanks for catching that